UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/Matrix.cs

```diff
@@ -105,7 +105,9 @@ public bool ContainsAtPlayer(Vector3Int position, GameObject gameObject)
         {
             if(Get<RegisterTile>(position, ObjectType.Player).Count() != 0)
             {
-                return Get<RegisterTile>(position, ObjectType.Player).First().gameObject.GetInstanceID() != gameObject.GetInstanceID();
+				GameObject otherPlayer = Get<RegisterTile>(position, ObjectType.Player).First().gameObject;
+
+				return !otherPlayer.GetComponent<PlayGroup.PlayerMove>().isGhost && otherPlayer.GetInstanceID() != gameObject.GetInstanceID();
```

Comment:
While this would work, it would be slow: `Get` and `GetComponent` calls should be minimized.
I didn't review #1095 properly back in the day and now I see how many expensive calls are there... Not your fault really
Will experiment a bit and try to come up with a more efficient solution, if I fail then I'll merge yours

Reply:
Ok, try ditching `ContainsAtPlayer` method completely and replacing `ContainsAt` with following:
```cs
        public bool ContainsAt(Vector3Int position, GameObject gameObject)
        {
            RegisterTile registerTile = gameObject.GetComponent<RegisterTile>();
            if ( !registerTile ) {
                return false;
            }
            
            if ( registerTile.ObjectType == ObjectType.Player ) 
            {
                var playersAtPosition = objects.Get<RegisterPlayer>(position);
                // makes sure it's not same player that called it
                if ( playersAtPosition.Count == 0 || playersAtPosition.Contains(registerTile) ) {
                    return false;
                }
                return playersAtPosition.First().IsBlocking;
            }
            var objectsAtPosition = objects.Get<RegisterTile>(position);
            return objectsAtPosition.Contains(registerTile);
        }
```
Worked nicely in a quick net test